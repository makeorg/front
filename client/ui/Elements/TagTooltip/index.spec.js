import React from 'react';
import renderer from 'react-test-renderer';
import { TagTooltip } from './index';

jest.mock('./style', () => ({
  TagsTooltipWrapperStyle: 'TagsTooltipWrapperStyle',
  TagsTooltipCrossStyle: 'TagsTooltipCrossStyle',
  TriangleUpStyle: 'TriangleUpStyle',
  TriangleDownStyle: 'TriangleDownStyle',
  LinkStyle: 'LinkStyle',
  TagsTooltipContainerStyle: 'TagsTooltipContainerStyle',
  TooltipSvgInfos: 'TooltipSvgInfos',
}));

jest.mock('Client/ui/Svg/elements', () => ({
  SvgClose: 'SvgClose',
}));

jest.mock('../AccessibilityElements', () => ({
  ScreenReaderItemStyle: 'ScreenReaderItemStyle',
}));

describe('TagTooltip', () => {
  it('must match the snapshot with default Props', () => {
    const component = renderer.create(<TagTooltip />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
