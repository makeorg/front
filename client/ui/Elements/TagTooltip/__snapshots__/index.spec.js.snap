// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagTooltip must match the snapshot with default Props 1`] = `
<TagsTooltipContainerStyle>
  <TriangleStyle />
  <TagsTooltipWrapperStyle>
    <ScreenReaderItemStyle>
      common.notifications.icons.information
    </ScreenReaderItemStyle>
    common.notifications.tags_filter
    <ScreenReaderItemStyle>
      consultation.tags.description
    </ScreenReaderItemStyle>
    <LinkStyle
      onClick={[Function]}
    >
      common.notifications.thank_you
    </LinkStyle>
    <TagsTooltipCrossStyle
      aria-label="common.notifications.icons.close"
      onClick={[Function]}
    >
      <SvgClose
        aria-hidden={true}
      />
    </TagsTooltipCrossStyle>
  </TagsTooltipWrapperStyle>
</TagsTooltipContainerStyle>
`;
